capstone/src/main/java/com/google/sps/servlets/ExploreServlet.java

```diff
@@ -14,9 +23,28 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Gson gson = new Gson();
-    ImmutableCollection<Club> clubs = PrototypeClubs.PROTOTYPE_CLUBS_MAP.values();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Query query = new Query(Constants.CLUB_PROP);
+    PreparedQuery results = datastore.prepare(query);
+    ImmutableList<Club> clubs =
+        Streams.stream(results.asIterable())
+            .limit(Constants.LOAD_LIMIT)
+            .map(entity -> createClubFromEntity(entity))
+            .collect(toImmutableList());
+
     String json = gson.toJson(clubs);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+  private Club createClubFromEntity(Entity entity) {
+    return new Club(
+        entity.getProperty(Constants.CLUB_NAME_PROP).toString(),
+        ImmutableList.copyOf((List<String>) entity.getProperty(Constants.MEMBER_PROP)),
+        ImmutableList.copyOf((List<String>) entity.getProperty(Constants.OFFICER_PROP)),
+        entity.getProperty(Constants.DESCRIP_PROP).toString(),
+        entity.getProperty(Constants.WEBSITE_PROP).toString(),
+        null);
```

Comment:
nit: can you add a comment showing what this parameter is? So readers know what is being set as null explicitly